Automattic/jetpack
# Connection: "set user token" endpoint

Adding new REST endpoint `/jetpack/v4/user-token` that allows us to add/update user tokens remotely.
To authenticate requests, we use the blog token.

#### Changes proposed in this Pull Request:
* new REST endpoint `/jetpack/v4/user-token`

#### Jetpack product discussion
1200327174179748-as-1200337741174716

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
1. Run unit tests on the `jetpack-connection` package, confirm there are no errors.
2. Use instructions in the 2920a-pb to setup the testing class in your sandbox.
3. Run the following command (replace `2` with the actual user ID): `$t->call_core_api( 'POST', '/jetpack/v4/user-token', array( 'user_token' => 'test.test.2' ) );`.
4. Use the "Broken Token" tool to confirm that user token has been added or updated.
5. Run the same command for a non-existing user ID: `$t->call_core_api( 'POST', '/jetpack/v4/user-token', array( 'user_token' => 'test.test.12345' ) );`.
6. Confirm that you receive the `invalid_argument_user_id` error.
